packages/assertion-monitor/index.ts

```diff
@@ -0,0 +1,200 @@
+// Updated processChunkedRange function
+async function processChunkedRange<T>(
+  fromBlock: bigint,
+  toBlock: bigint,
+  chunkSize: bigint,
+  client: PublicClient,
+  processChunk: ChunkProcessFunction<T>
+): Promise<T[]> {
+  const results: T[] = []
+
+  if (fromBlock === toBlock) {
+    const result = await processChunk(fromBlock, toBlock, client)
+    results.push(result)
+    return results
+  }
+
+  let currentFromBlock = fromBlock
+
+  while (currentFromBlock <= toBlock) {
+    const currentToBlock =
+      currentFromBlock + chunkSize - 1n < toBlock
+        ? currentFromBlock + chunkSize - 1n
+        : toBlock
+
+    const result = await processChunk(currentFromBlock, currentToBlock, client)
+    results.push(result)
+
+    if (currentToBlock === toBlock) break
+
+    currentFromBlock = currentToBlock + 1n
+  }
+
+  return results
+}
+const getBlockRange = async (
+  client: PublicClient,
+  childChainInfo: ChainInfo
+) => {
+  const latestBlockNumber = await client.getBlockNumber()
+  const toBlock = BigInt(options.toBlock) || latestBlockNumber
+
+  if (options.fromBlock) {
+    return { fromBlock: BigInt(options.fromBlock), toBlock }
+  }
+
+  const blockTime = getDefaultBlockRange(
+    getChainFromId(childChainInfo.parentChainId)
+  )
+
+  const fromBlock = await client.getBlock({
+    blockNumber: latestBlockNumber - BigInt(blockTime),
+  })
+
+  return { fromBlock: fromBlock.number, toBlock }
+}
+
+const monitorNodeCreatedEvents = async (childChainInfo: ChainInfo) => {
+  const parentChain = getChainFromId(childChainInfo.parentChainId)
+  const client = createPublicClient({
+    chain: parentChain,
+    transport: http(childChainInfo.parentRpcUrl),
+  })
+
+  const { fromBlock, toBlock } = await getBlockRange(client, childChainInfo)
+
+  const getLogsForChunk = async (
+    chunkFromBlock: bigint,
+    chunkToBlock: bigint,
+    client: PublicClient
+  ) => {
+    return client
+      .getLogs({
+        address: childChainInfo.ethBridge.rollup as `0x${string}`,
+        event: nodeCreatedEventAbi,
+        fromBlock: chunkFromBlock,
+        toBlock: chunkToBlock,
+      })
+      .catch(console.error)
+  }
+
+  const logsArray = await processChunkedRange(
+    fromBlock,
+    toBlock,
+    CHUNK_SIZE,
+    client,
+    getLogsForChunk
+  )
+
+  const logs = logsArray.flat()
+
+  const isDefaultSettings = !options.fromBlock && !options.toBlock
+  const durationString = isDefaultSettings
+    ? 'in last 7 days'
+    : `from block ${fromBlock} to block ${toBlock}`
+
+  const latestSafeBlock = await client.getBlock({ blockTag: 'safe' })
+
+  const latestSafeBlockTimestamp =
+    new Date(Number(latestSafeBlock.timestamp) * 1000).toLocaleString() + ' UTC'
+
+  const isLatestSafeBlockWithinRange =
+    latestSafeBlock.number < toBlock && latestSafeBlock.number > fromBlock
+
+  if (!logs || logs.length === 0) {
+    return {
+      chainName: childChainInfo.name,
+      alertMessage: `No assertion events found on ${
```

Comment:
I'd be a bit more explicit: `No assertion creation events...` . Just to explicitly mention that we are not tracking assertion confirmation events.